src/mds/Server.cc

```diff
@@ -2642,7 +2642,8 @@ void Server::dispatch_client_request(MDRequestRef& mdr)
 	mdr->batch_op_map->erase(it);
 	return;
       }
-      mdr = std::move(new_batch_head);
+      mds->finisher->queue(new C_MDS_RetryRequest(mdcache, new_batch_head));
```

Comment:
Doesn't find_new_head() does the same ->killed checks?

Reply:
Ah, you are right. In that case, this is just a cleanup so that `mdr` can just remain constant.

Reply:
Because of that, I'll merge this commit with #54484.